pkg/collector/reconcile/statefulset_test.go

```diff
@@ -138,7 +139,45 @@ func TestExpectedStatefulsets(t *testing.T) {
 		oldSs := collector.StatefulSet(param.Config, logger, param.Instance)
 		oldSs.Spec.Selector.MatchLabels["app.kubernetes.io/version"] = "latest"
 		oldSs.Spec.Template.Labels["app.kubernetes.io/version"] = "latest"
-		oldSs.Name = "update-ss"
+		oldSs.Name = "update-selector"
+
+		err := expectedStatefulSets(context.Background(), param, []v1.StatefulSet{oldSs})
+		assert.NoError(t, err)
+		exists, err := populateObjectIfExists(t, &v1.StatefulSet{}, types.NamespacedName{Namespace: "default", Name: oldSs.Name})
+		assert.NoError(t, err)
+		assert.True(t, exists)
+
+		newSs := collector.StatefulSet(param.Config, logger, param.Instance)
+		newSs.Name = oldSs.Name
+		err = expectedStatefulSets(context.Background(), param, []v1.StatefulSet{newSs})
+		assert.NoError(t, err)
+		exists, err = populateObjectIfExists(t, &v1.StatefulSet{}, types.NamespacedName{Namespace: "default", Name: oldSs.Name})
+		assert.NoError(t, err)
+		assert.False(t, exists)
+
+		err = expectedStatefulSets(context.Background(), param, []v1.StatefulSet{newSs})
+		assert.NoError(t, err)
+		actual := v1.StatefulSet{}
+		exists, err = populateObjectIfExists(t, &actual, types.NamespacedName{Namespace: "default", Name: oldSs.Name})
+		assert.NoError(t, err)
+		assert.True(t, exists)
+		assert.Equal(t, newSs.Spec.Selector.MatchLabels, actual.Spec.Selector.MatchLabels)
+	})
+
+	t.Run("change Spec.VolumeClaimTemplates should recreate statefulset", func(t *testing.T) {
```

Comment:
where is the volume claim template changed in this test? It seems we just make a collector once with a volume claim template, but never change it to see the recreation?

Reply:
So we first create STS _with_ volume claim templates and then create a new STS _without_ one, in which case we should also recreate the STS. But it might be confusing for testing purposes, I made it clearer by switching it around (first create STS with no volume claim templates, then add one).